app/controllers/provider_interface/conditions_controller.rb

```diff
@@ -22,10 +23,18 @@ def update
       redirect_to action: :edit unless @conditions_form.valid?
 
       if @conditions_form.conditions_met?
-        ConfirmOfferConditions.new(application_choice: @application_choice).save || raise('ConfirmOfferConditions failure')
+        ConfirmOfferConditions.new(
+          actor: current_provider_user,
+          application_choice: @application_choice,
+        ).save || raise('ConfirmOfferConditions failure')
```

Comment:
I know it's not part of this PR, but this should raise a more informative error - this won't tell us what has gone wrong (an invalid state transition I think).

Reply:
It seems to be a pattern in many services corresponding to provider actions to `.save` instead of `.save!` and rescue `Workflow::NoTransitionAllowed`, returning `false`. We could refactor these services as a separate PR. If these services raise a `ProviderInterface::NotAuthorisedError` now that we check `make_decisions`, I think we'll be able to see it.